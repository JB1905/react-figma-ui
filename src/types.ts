import { colorNames, iconNames, sizes, tints, weights } from './constants';

export type ColorName = (typeof colorNames)[number];
export type IconName = (typeof iconNames)[number];
export type Size = (typeof sizes)[number];
export type Tint = (typeof tints)[number];
export type Weight = (typeof weights)[number];
