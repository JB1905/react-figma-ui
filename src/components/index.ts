export * from './Button';
export * from './Checkbox';
export * from './Disclosure';
export * from './Icon';
export * from './IconButton';
export * from './Input';
export * from './Label';
export * from './Onboarding';
export * from './Radio';
export * from './SectionTitle';
export * from './SelectMenu';
export * from './Switch';
export * from './Textarea';
export * from './Type';
