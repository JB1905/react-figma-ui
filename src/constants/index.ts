export * from './colorNames';
export * from './iconNames';
export * from './sizes';
export * from './tints';
export * from './weights';
