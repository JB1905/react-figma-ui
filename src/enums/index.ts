export { Size } from './Size';
export { Tint } from './Tint';
export { Weight } from './Weight';
