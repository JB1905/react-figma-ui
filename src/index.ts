export { Button } from './components/Button';
export { Checkbox } from './components/Checkbox';
export { Disclosure } from './components/Disclosure';
export { Icon } from './components/Icon';
export { IconButton } from './components/IconButton';
export { Input } from './components/Input';
export { Label } from './components/Label';
export { Onboarding } from './components/Onboarding';
export { Radio } from './components/Radio';
export { SectionTitle } from './components/SectionTitle';
export { SelectMenu } from './components/SelectMenu';
export { Switch } from './components/Switch';
export { Textarea } from './components/Textarea';
export { Type } from './components/Type';
