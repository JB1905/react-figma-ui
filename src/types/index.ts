import { colorNames } from '../constants/colorNames';
import { iconNames } from '../constants/iconNames';
import { sizes } from '../constants/sizes';
import { tints } from '../constants/tints';
import { weights } from '../constants/weights';

export type ColorName = typeof colorNames;
export type IconName = typeof iconNames;
export type Size = typeof sizes;
export type Tint = typeof tints;
export type Weight = typeof weights;
