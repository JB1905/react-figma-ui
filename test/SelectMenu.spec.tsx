import React from 'react';
import { render } from '@testing-library/react';

import { SelectMenu, SelectMenuOption } from '../src';

describe('SelectMenu', () => {
  const options = [
    { value: 1, label: 'Option 1' },
    { value: 2, label: 'Option 2' },
    { value: 3, label: 'Option 3' },
  ];

  it('should render SelectMenu', () => {
    const { container } = render(
      <SelectMenu
        options={options}
        render={({ value, label }) => (
          <SelectMenuOption value={value} key={value}>
            {label}
          </SelectMenuOption>
        )}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should render SelectMenu with custom className', () => {
    const { container } = render(
      <SelectMenu
        options={options}
        className="custom-class-name"
        render={({ value, label }) => (
          <SelectMenuOption value={value} key={value}>
            {label}
          </SelectMenuOption>
        )}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
