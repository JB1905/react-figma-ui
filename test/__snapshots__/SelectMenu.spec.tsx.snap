// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectMenu should render SelectMenu 1`] = `
<div>
  <select
    class="select-menu"
  >
    <option
      value="1"
    >
      Option 1
    </option>
    <option
      value="2"
    >
      Option 2
    </option>
    <option
      value="3"
    >
      Option 3
    </option>
  </select>
</div>
`;

exports[`SelectMenu should render SelectMenu with custom className 1`] = `
<div>
  <select
    class="select-menu custom-class-name"
  >
    <option
      value="1"
    >
      Option 1
    </option>
    <option
      value="2"
    >
      Option 2
    </option>
    <option
      value="3"
    >
      Option 3
    </option>
  </select>
</div>
`;
